### PR TITLE
Implement router with go-restful.

### PR DESCRIPTION
**Pull request by calavera**
It replaces the old router with a very similar one using go-restful.
It doesn't define any swagger documentation yet. The purpose of this change is
to provide the basic infrastucture to migrate to a swagger defined api, but
we'll add the definitions in consequent pull requests.

Disclaimer: This is very rough in the edges.

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by SvenDowideit** [comment]
\o/

**Comment by LK4D4** [comment]
Do we still need mux?

File: hack/vendor.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ clone git github.com/Sirupsen/logrus v0.8.2 # logrus is a common dependency amon
 clone git github.com/docker/libtrust 9cbd2a1374f46905c68a4eb3694a130610adc62a
 clone git github.com/go-check/check 64131543e7896d5bcc6bd5a76287eb75ea96c673
 clone git github.com/gorilla/context 14f550f51a
+clone git github.com/emicklei/go-restful 8fe9edc3345c61dc3b82a69dd75e053f70bac253
 clone git github.com/gorilla/mux e444e69cbd
```
Comment (jfrazelle) [inline hack/vendor.sh]: can we remove mux then?

Comment (calavera) [inline hack/vendor.sh]: I think so. I need to double check that it's not used anywhere in the stack, but I'm planning on removing it.

Comment (jfrazelle) [inline hack/vendor.sh]: +1

**Comment by calavera** [comment]
@LK4D4, @jfrazelle we depend on this change in distribution to be able to remove mux from the dependencies:

https://github.com/docker/distribution/pull/955

**Comment by stevvooe** [comment]
I'm a little concerned about replacing a small routing library with a large web framework that replaces all of the handler signatures. Is there not a sufficient request router that doesn't depart from the `net/http` package?

**Comment by stevvooe** [comment]
I'm a little concerned about replacing a small routing library with a large web framework that replaces all of the handler signatures. Is there not a sufficient request router that doesn't depart from the `net/http` package?

**Comment by stevvooe** [comment]
Now that I look at this closer, I am a big :-1: on go-restful.

**Comment by calavera** [comment]
Closing, I don't like go-restful either.

**Pull request closed without merging**